provider-service/base/Makefile

```diff
@@ -0,0 +1,24 @@
+##@ Development
+unit-test:
+	go test ./... -tags=unit
+
+decoupled-test:
+	go test ./... -tags=decoupled
+
+test: unit-test decoupled-test integration-test
+
+integration-test:
+	docker compose up -d
+	$(MAKE) wait-for-pubsub
+	go test ./... -v -tags=integration
+	docker compose down
+
+# Wait for the Pub/Sub emulator to become healthy
```

Comment:
```suggestion
```
Think it's fine without this comment. Implementation and naming is self-explanatory. Should reserve the comment for why (if it's unclear), but I think it is clear anyway